packages/autocertifier/src/RestServer.ts

```diff
@@ -0,0 +1,268 @@
+
+            sendResponse(res, certifiedSubdomain)
+        } catch (err) {
+            sendError(res, err)
+            return
+        }
+    }
+
+    private updateSubdomainIpAndPort = async (req: express.Request, res: express.Response): Promise<void> => {
+        const subdomain = req.params.subdomain
+        const body = req.body as UpdateIpAndPortRequest
+
+        if (!body || !body.streamrWebSocketPort) {
+            const err = new SteamrWebSocketPortMissing('Streamr websocket port not given')
+            sendError(res, err)
+            return
+        }
+        const streamrWebSocketPort = req.body.streamrWebSocketPort + ''
+
+        if (!body || !body.token) {
+            const err = new TokenMissing('Token not given')
+            sendError(res, err)
+            return
+        }
+        const token = body.token
+        const sessionId = body.sessionId
+        
+        const ipAndPort = extractIpAndPort(req)
+        if (!ipAndPort) {
+            const err = new FailedToExtractIpAddress('Failed to extract IP address from request')
+            sendError(res, err)
+            return
+        }
+        logger.debug('updateSubdomainIpAndPort() '
+            + 'subdomain: ' + subdomain + ', ip: ' + ipAndPort.ip
+            + ', port: ' + ipAndPort.port + ', streamrWebSocketPort: ' + streamrWebSocketPort
+            + ', sessionId: ' + ' ' + sessionId + ', token: ' + token)
+        try {
+            await this.engine.updateSubdomainIpAndPort(subdomain, ipAndPort.ip,
+                ipAndPort.port, streamrWebSocketPort, sessionId, token)
+
+            sendResponse(res)
+        } catch (err) {
+            sendError(res, err)
+        }
+    }
+
+    // TODO: use async fs methods?
+    private createSelfSignedCertsIfTheyDontExist(): void {
+        if (!fs.existsSync(this.caCertPath) || !fs.existsSync(this.caKeyPath) ||
+            !fs.existsSync(this.certPath) || !fs.existsSync(this.keyPath)) {
+            const certs = createSelfSignedCertificate(this.ownFqdn, 1200)
+            if (!fs.existsSync(path.dirname(this.caCertPath))) {
+                fs.mkdirSync(path.dirname(this.caCertPath), { recursive: true })
+            }
+            if (!fs.existsSync(path.dirname(this.caKeyPath))) {
+                fs.mkdirSync(path.dirname(this.caKeyPath), { recursive: true })
+            }
+            if (!fs.existsSync(path.dirname(this.certPath))) {
+                fs.mkdirSync(path.dirname(this.certPath), { recursive: true })
+            }
+            if (!fs.existsSync(path.dirname(this.keyPath))) {
+                fs.mkdirSync(path.dirname(this.keyPath), { recursive: true })
+            }
+            fs.writeFileSync(this.caCertPath, certs.caCert, { flag: 'w' })
+            fs.writeFileSync(this.caKeyPath, certs.caKey, { flag: 'w' })
+            fs.writeFileSync(this.certPath, certs.serverCert, { flag: 'w' })
+            fs.writeFileSync(this.keyPath, certs.serverKey, { flag: 'w' })
```

Comment:
Introduce a helper to avoid code duplication

Reply:
added todo comment